File: src/checkIfPangram.js
/**
 * https://leetcode.com/problems/check-if-the-sentence-is-pangram/submissions/
 * Runtime: 108 ms, faster than 17.35% of JavaScript online submissions for Check if the Sentence Is Pangram.
 * Memory Usage: 38.3 MB, less than 91.78% of JavaScript online submissions for Check if the Sentence Is Pangram.
 * @param {string} sentence
 * @return {boolean}
 */
var checkIfPangram = function (sentence) {
  let is_pangram = true;
  let letters = [..."abcdefghijklmnopqrstuvwxyz"];
  letters.forEach((letter) => {
    if (!sentence.includes(letter)) {
      is_pangram = false;
    }
  });
  return is_pangram;
};

File: src/check_if_pangram.rs
///
/// https://leetcode.com/problems/check-if-the-sentence-is-pangram/
pub fn check_if_pangram(word: &str) -> bool {
    let alphabet = "abcdefghijklmnopqrstuvwxyz";
    let mut is_pangram = true;
    for alpha in alphabet.chars() {
        if !word.contains(alpha) {
            is_pangram = false;
        }
    }
    is_pangram
}
#[cfg(test)]
mod tests {
    use super::check_if_pangram;
    #[test]
    fn test_check_if_pangram() {
        assert_eq!(
            check_if_pangram("thequickbrownfoxjumpsoverthelazydog"),
            true
        );
        assert_eq!(check_if_pangram("leetcode"), false);
    }
}
